src/remake.jl

```diff
@@ -69,12 +69,9 @@ function remake(prob::ODEProblem; f = missing,
             u0 = prob.u0
         end
         if (eltype(p) <: Pair && !isempty(p)) || (eltype(u0) <: Pair && !isempty(u0)) # one is a non-empty symbolic map
-            hasproperty(prob.f, :sys) && hasfield(typeof(prob.f.sys), :ps) ||
+            has_sys(prob.f) ||
```

Comment:
Let's solve https://github.com/SciML/DifferentialEquations.jl/issues/922 at the same time here. If it doesn't have a system, then we shouldn't care what `p` is an just use it. `p` in a general diffeq is allowed to be a dictionary or Pair thing if people want, we shouldn't care.

But also, let's add a keyword argument `interpret_symbolicmap = true` and change the error message saying that this can be bypassed by setting that to false.